samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.CreateBookingCommand.g.cs

```diff
@@ -70,22 +70,50 @@ public virtual async Task<BookingKeyDto> Handle(CreateBookingCommand request, Ca
 		OnExecuting(request);
 
 		var entityToCreate = _entityFactory.CreateEntity(request.EntityDto);
-		if(request.EntityDto.BookingForCustomer is not null)
+		if(request.EntityDto.BookingForCustomerId is not null)
+		{
+			var relatedKey = CreateNoxTypeForKey<Customer, Nox.Types.AutoNumber>("Id", request.EntityDto.BookingForCustomerId);
+			var relatedEntity = await _dbContext.Customers.FindAsync(relatedKey);
+			if(relatedEntity is not null && relatedEntity.DeletedAtUtc == null)
```

Comment:
we can remove the `relatedEntity.DeletedAtUtc == null` validationm ticket #890 will solve this
thks

regarding your question if the entity does not exists we should throw an exception, do create a specific exception to throw , for example: RelatedEntityNotFoundException(request.EntityDto.BookingForCustomerId)

Reply:
Updated for both points